File: backend/app/schemas/notification.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel

NotificationScope = Literal["dashboard", "audit", "all"]


class NotificationRead(BaseModel):
    id: str
    title: str
    description: str
    created_at: datetime
    scope: Literal["dashboard", "audit"]
    kind: str
    target_id: str | None = None

File: frontend/src/types/notification.ts
export type NotificationScope = 'dashboard' | 'audit' | 'all';

export interface AppNotification {
  id: string;
  title: string;
  description: string;
  created_at: string;
  scope: 'dashboard' | 'audit';
  kind: string;
  target_id: string | null;
}
